Examples/Python/src/Obj.cpp

```diff
@@ -47,6 +48,47 @@ void addObj(Context& ctx) {
               Acts::GeometryView3D view3D;
               Acts::ObjVisualization3D obj;
 
+              for (const auto& surface : surfaces) {
+                view3D.drawSurface(obj, *surface, viewContext,
+                                   Acts::Transform3::Identity(), sConfig);
+              }
+              obj.write(fileName);
+            });
+    mex.def("writeVolumesObj",
+            [](const std::vector<std::shared_ptr<Experimental::DetectorVolume>>&
+                   Volumes,
+               const GeometryContext& viewContext,
+               const std::array<int, 3>& viewRgb, unsigned int viewSegements,
+               const std::string& fileName) {
+              Acts::ViewConfig sConfig = Acts::ViewConfig{viewRgb};
+              sConfig.nSegments = viewSegements;
+              Acts::GeometryView3D view3D;
+              Acts::ObjVisualization3D obj;
+
+              for (const auto& volume : Volumes) {
+                view3D.drawDetectorVolume(obj, *volume, viewContext,
+                                          Acts::Transform3::Identity(),
+                                          sConfig);
+              }
+              obj.write(fileName);
+            });
+    mex.def("writeVolumesSurfacesObj",
+            [](const std::vector<std::shared_ptr<Surface>>& surfaces,
```

Comment:
I think this is resolved by caching the surfaces even if they are contained by the volumes. My guess as to why this is necessary, is that something in the obj conversion works by reference and somehow relies on the initial memory address of the surface. But ether way while converting all MDTs with the tubes at once it doesn't seem to make a difference memory wise whether or not I cache the surfaces.

Reply:
Talking about it with @paulgessinger we concluded that this is a deep rooted "issue" which is not really worth addressing right now since it doesn't appear to cost much (if anything).